### PR TITLE
[MB-6845] Update Customer App routes

### PR DESCRIPTION
**Pull request by suzubara**
## Description

This PR cleans up some of the routing in the customer app. Some paths changed based on input from the design team.

- Most paths are now defined in a new `constants/routes.js` file
- Use the [`generatePath`](https://reactrouter.com/web/api/generatePath) helper in order to interpolate variables into path string constants
- Removed `serviceMemberId` parameter from customer paths
- Added some route tests to `MyMove/index.test.jsx` using RTL
- Deleted deprecated routes `/sm_style_guide` and `/dps_cookie` (and related code)
- There are some paths that may still change, or that I didn't create constants for, because they are in areas of the app that I know we're going to be rebuilding or otherwise modifying (ie, the new Profile pages, the PPM pages).

Paths that changed are:
```
/privacy-and-security-policy -> /privacy-security

/service-member/:serviceMemberId/conus-status           -> /service-member/conus-oconus
/service-member/:serviceMemberId/create                 -> /service-member/dod-info
/service-member/:serviceMemberId/name                   -> /service-member/name
/service-member/:serviceMemberId/contact-info           -> /service-member/contact-info
/service-member/:serviceMemberId/duty-station           -> /service-member/current-duty
/service-member/:serviceMemberId/residence-address      -> /service-member/current-address
/service-member/:serviceMemberId/backup-mailing-address -> /service-member/backup-address
/service-member/:serviceMemberId/backup-contacts        -> /service-member/backup-contact

/orders -> /orders/info

/moves/:moveId/select-type                                -> /moves/:moveId/shipment-type
/moves/:moveId/mto-shipments/:mtoShipmentId/edit-shipment -> /moves/:moveId/shipments/:mtoShipmentId/edit
```

## Reviewer Notes

Primary thing to test is that the above URLs work as expected, and the usual customer flows are functional in the app. I did not add any redirects because I don't believe we're linking to any of these pages from outside of the app (such as from an email) but please let me know if that's wrong.

## Setup

To run the app locally:
```sh
make server_run
make client_run
```

## Code Review Verification Steps

* [ ] If the change is risky, it has been tested in experimental before merging.
* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#logging)
* [ ] The requirements listed in [Querying the Database Safely](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#querying-the-database-safely) have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](https://github.com/transcom/mymove/wiki/migrate-the-database#zero-downtime-migrations))
  * [ ] Have been communicated to #g-database
  * [ ] Secure migrations have been tested following the instructions in our [docs](https://github.com/transcom/mymove/wiki/migrate-the-database#secure-migrations)
* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-6845) for this change

### CODE REVIEW

**Review by Shkeating — APPROVED** [review]
all urls are changed as noted ✅

**Comment by Shkeating** [comment]
I scrolled down and noticed there were happo changes, I am seeing some odd errors in there that you might want to check out

**Comment by suzubara** [comment]
> I scrolled down and noticed there were happo changes, I am seeing some odd errors in there that you might want to check out

thanks for noticing!! they should be fixed now

File: src/pages/MyMove/Home/index.test.jsx [hunk 1]
```diff
@@ -43,14 +43,13 @@ const defaultProps = {
...
+    status: 'DRAFT',
+  },
   uploadedOrderDocuments: [],
 };
 
-const mountHome = (props = {}) => {
-  return mount(<Home {...defaultProps} {...props} />);
-};
-
 const mountHomeWithProviders = (props = {}) => {
```
Comment (kimallen) [inline src/pages/MyMove/Home/index.test.jsx:53]: What are Providers?

File: src/constants/routes.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  ACCESSIBILITY_PATH: '/accessibility',
+};
+
+export const customerRoutes = {
+  ACCESS_CODE_PATH: '/access-code',
+  CONUS_OCONUS_PATH: '/service-member/conus-oconus',
+  DOD_INFO_PATH: '/service-member/dod-info',
+  NAME_PATH: '/service-member/name',
+  CONTACT_INFO_PATH: '/service-member/contact-info',
+  CURRENT_DUTY_STATION_PATH: '/service-member/current-duty',
```
Comment (kimallen) [inline src/constants/routes.js:14]: I notice that these paths are different than the ones in the [spreadsheet](https://docs.google.com/spreadsheets/d/1VWtagZWTcmgqfWaLdQlUHKBJJVtqBXlEJ_KuMMOSZMs/edit#gid=0) of URLs.  Are these here the result of later conversations, and the spreadsheet has not been updated?

**Review by kimallen — COMMENTED** [review]
I didn't get to testing the app yet, but the code looks good!   I like how you pulled the routs out into constants.  Thanks also for putting a detailed description- it makes review easier.

File: src/pages/MyMove/Home/index.test.jsx (hunk 1 above)
Comment (suzubara) [reply src/pages/MyMove/Home/index.test.jsx:53]: Providers / containers are the term used to describe high-level wrappers that "provide" data or context to the app, so in this case that's the Redux store & routing context.

File: src/constants/routes.js (hunk 2 above)
Comment (suzubara) [reply src/constants/routes.js:14]: Ah yes, sorry for the lack of clarification! I went over the URLs with James and the results of that conversation are on the 2nd tab of the spreadsheet (which lists all Customer App URLs, including some that may change but not as part of this branch)

File: src/pages/MyMove/Home/index.test.jsx (hunk 1 above)
Comment (suzubara) [reply src/pages/MyMove/Home/index.test.jsx:53]: to give context to this change, originally I was still using `mountHome` to write tests for the Home component without relying on data from the providers, but eventually ended up using `mountHomeWithProviders` for all of the tests, so `mountHome` was no longer necessary

**Review by kimallen — APPROVED** [review]
The routes listed look good and worked fine.  Approving, with some questions/requests below.
2 questions:
Do you want to include `review/edit-orders` in this too? or do you expect changes later to that page?
Since we are waiting to change some of the routes for Profile and PPM, can you make sure that's explicitly documented in another story (or existing stories)?
Can you orJames update the spreadsheet so that Sheet 1 doesn't have the outdated paths anymore?

**Comment by suzubara** [comment]
> Do you want to include review/edit-orders in this too? or do you expect changes later to that page?

Since we're about to update the Edit pages I grouped this page with that work.

>Since we are waiting to change some of the routes for Profile and PPM, can you make sure that's explicitly documented in another story (or existing stories)?

Yes!

>Can you orJames update the spreadsheet so that Sheet 1 doesn't have the outdated paths anymore?

I renamed the tabs so it's clearer Sheet 1 was the first draft

**Pull request merged**